src/api/layer.js

```diff
@@ -42,6 +42,7 @@ export default class Layer {
     }
 
     _init(id, source, style) {
+        source = source._clone();        
```

Comment:
I think this line is not needed (extra _clone)

Reply:
Why? the constructor doesn't clone it, and there can be two layers with the same source passed in both cases through the constructor

Reply:
I think this is done in `this.update(source, style);` at the end of the `_init` function

Reply:
Yep, you're right